# Tired of Cliché Datasets? Here Are 20 Awesome Alternatives For Any Task
## TODO
![](images/unsplash.jpg)
<figcaption style="text-align: center;">
    <strong>
        Photo by 
        <a href='https://unsplash.com/@dogukan?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'> Doğukan Şahin</a>
        on 
        <a href='https://unsplash.com/s/photos/tired?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText'>Pexels.</a> All images are by the author unless specified otherwise.
    </strong>
</figcaption>

### Setup

In [7]:
import warnings

import gdown
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
from matplotlib import rcParams

rcParams["xtick.labelsize"] = 15
rcParams["ytick.labelsize"] = 15

warnings.filterwarnings("ignore")

"I'm going to puke over my RGB backlit-keyboard *so hard* if I see one more person using Titanic, Iris, Wine or Boston datasets!"

This is the feeling you might gradually develop after being a data science learner for a while. You just can't help it - everyone wants the easy thing. Beginners use these datasets because they are stupidly straightforwad; most course creaters and bloggers use them because they are just one single Google search away.

# Regression datasets

## 1. Diamond prices and carat regression

My favorite from this list is the diamonds dataset. It is ideal in length (+50k samples) and have multiple targets you can predict as a regression or a multi-class classification task:

In [4]:
diamonds = sns.load_dataset("diamonds").sample(10).reset_index(drop=True)
diamonds

carat        cut color clarity  depth  table  price     x     y     z
0   0.90  Very Good     F     VS1   61.2   56.0   5324  6.18  6.23  3.80
1   0.41       Good     J     VS2   64.0   59.0    732  4.71  4.64  2.99
2   0.92  Very Good     H     VS1   62.9   58.0   4228  6.17  6.20  3.89
3   0.24    Premium     H    VVS1   61.2   58.0    432  3.96  4.01  2.44
4   0.30    Premium     E     VS2   59.2   59.0    844  4.41  4.37  2.60
5   0.93  Very Good     F     VS2   62.7   58.0   4925  6.21  6.27  3.91
6   0.41    Premium     J    VVS1   62.2   59.0    994  4.75  4.71  2.94
7   0.41    Premium     G     VS1   60.5   58.0    899  4.76  4.80  2.89
8   0.30      Ideal     G    VVS1   62.0   55.0   1013  4.34  4.31  2.68
9   0.71  Very Good     H     VS2   61.8   58.0   2368  5.66  5.70  3.51

In [5]:
sns.load_dataset("diamonds").shape

(53940, 10)

**🎯 Targets: 'carat', 'price' (regression); 'cut', 'color', 'clarity' (multi-class)**

**🔗 Link: [Kaggle](https://www.kaggle.com/shivam2503/diamondshttps://www.kaggle.com/shivam2503/diamonds)**

**📦Dimensions: (53940, 10)**

**⚙Missing values: No**

**[📚Starter notebook](https://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysishttps://www.kaggle.com/fuzzywizard/diamonds-in-depth-analysis)**

## 2. Age of Abalone shells

This is a rather unique dataset from the field of zoology. The task is to predict the age of Abalone shells (a type of mollusc) using several physical measurements. Traditionally, their age are found by cutting through their cone, staining them and counting the number of rings inside the shell under a microscope. 

For zooligists, this might be fun but for data scientists, not so much:

Signature:
gdown.download(
    url,
    output=None,
    quiet=False,
    proxy=None,
    speed=None,
    use_cookies=True,
)
Docstring:
Download file from URL.

Parameters
----------
url: str
    URL. Google Drive URL is also supported.
output: str, optional
    Output filename. Default is basename of URL.
quiet: bool
    Suppress terminal output. Default is False.
proxy: str
    Proxy.
speed: float
    Download byte size per second (e.g., 256KB/s = 256 * 1024).
use_cookies: bool
    Flag to use cookies. Default is True.

Returns
-------
output: str
    Output filename.
File:      c:\users\user\anaconda3\envs\medium_articles\lib\site-packages\gdown\download.py
Type:      function
